### PR TITLE
Update open api client to latest open api spec

### PR DESCRIPTION
**Pull request by frankisawesome**
# Summary

- New primary sale related types and endpoints
- Description and API base update to be up to date
- Some functions are rolled forward with description changes or additional functionality
- One endpoint (create project) is deleted from the open api spec and all functions and examples are removed (confirmed with hub team)

# Why the changes

# Things worth calling out

# Before merging
- [ ] ***For Immutable developers:*** Do the changes in this PR require documentation updates? Please refer to this [SDK documentation guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide) and list links to documentation update PRs (they don't have to be merged) below (or write `N/A`):
    - [ ] *Add documentation update 1*
    - [ ] *Add documentation update 2*

### CODE REVIEW

File: src/api/models/create-project-group-request.ts
Comment (lfportal) [inline src/api/models/create-project-group-request.ts:1]: **Question:** Given "create project" was removed, should this file still exist?

File: src/api/models/create-project-request.ts
Comment (lfportal) [inline src/api/models/create-project-request.ts:1]: **Question:** Similar, given "create project" was removed, should this file still exist?

File: src/api/models/create-project-response.ts
Comment (lfportal) [inline src/api/models/create-project-response.ts:1]: **Question:** Similar, given "create project" was removed, should this file still exist?

**Review by lfportal — APPROVED** [review]

File: src/api/models/create-project-group-request.ts
Comment (frankisawesome) [reply src/api/models/create-project-group-request.ts:1]: discussed offline

**Review by CodeSchwert — APPROVED** [review]
Majority of the changes was just adding `Immutable X API` as a description on auto-generated files! Wish that could have been moved to a separate PR 😅

**Pull request merged**